app/models/miq_expression.rb

```diff
@@ -575,7 +575,7 @@ def self._to_ruby(exp, context_type, tz)
   def to_sql(tz = nil)
     tz ||= "UTC"
     @pexp, attrs = preprocess_for_sql(@exp.deep_clone)
-    sql = to_arel(@pexp, tz).to_sql if @pexp.present?
+    sql = Expression.build(@pexp, tz).to_sql if @pexp.present?
```

Comment:
Ageed, RE: renaming instance variables such as `@exp`... we have to be very careful because MiqExpression objects are serialized with their instance variables so we'd need to ensure we dump/load from either format.